Nimble/CallRecorder.swift

```diff
@@ -0,0 +1,312 @@
+
+/* bug in Swift causes every enum WITHOUT an associated value's "description" to be the first declared
+enum value WITHOUT an associated value's description
+i.e. -> since ".DontCare" is the first enum value then the default description for ".DontCare",
+".NonNil", and ".Nil" will all be "DontCare" -> must override to fix issue by conforming to "CustomStringConvertible" 
+    This is not reproducable in a Playground */
+
+public enum Argument : CustomStringConvertible {
+    case DontCare
+    case NonNil
+    case Nil
+    case InstanceOf(type: Any.Type)
+    case InstanceOfWith(type: Any.Type, option: ArgumentOption)
+    case KindOf(type: AnyObject.Type)
+    
+    public var description: String {
+        switch self {
+        case .DontCare:
+            return "Argument.DontCare"
+        case .NonNil:
+            return "Argument.NonNil"
+        case .Nil:
+            return "Argument.Nil"
+        case .InstanceOf(let type):
+            return "Argument.InstanceOf(\(type))"
+        case .InstanceOfWith(let input):
+            return "Argument.InstanceOfWith(\(input.type), \(input.option))"
+        case .KindOf(let type):
+            return "Argument.KindOf(\(type))"
+        }
+    }
+}
+
+public enum ArgumentOption : CustomStringConvertible {
+    case DontCare
+    case NonOptional
+    case Optional
+    
+    public var description: String {
+        switch self {
+        case .DontCare:
+            return "ArgumentOption.DontCare"
+        case .NonOptional:
+            return "ArgumentOption.NonOptional"
+        case .Optional:
+            return "ArgumentOption.Optional"
+        }
+    }
+}
+
+public struct DidCallResult {
+    public let success: Bool
+    public let recordedCallsDescription: String
+//    public let successfullyCalledCount: Int
```

Comment:
No, sorry. This is a note to myself. It was a reminder for me to see if I wanted to include the called count of a function when the tester is wanting a specific 'count' times called (also 'atLeast' and 'atMost'). However after thinking about it I feel like it would be too much information since I already print out every called function on the object at the time it is tested (if the test fails)